common/src/main/java/tachyon/util/io/ByteIOUtils.java

```diff
@@ -0,0 +1,256 @@
+   *
+   * @param buf output byte array
+   * @param pos offset into the byte buffer to write
+   * @param v int value to write
+   */
+  public static void writeInt(byte[] buf, int pos, int v) {
+    checkBoundary(buf, pos, 4);
+    buf[pos ++] = (byte) (0xff & (v >> 24));
+    buf[pos ++] = (byte) (0xff & (v >> 16));
+    buf[pos ++] = (byte) (0xff & (v >> 8));
+    buf[pos] = (byte) (0xff & v);
+  }
+
+  /**
+   * Writes a specific integer value (4 bytes) to the output byte buffer at given offset.
+   *
+   * @param buf output byte buffer
+   * @param pos offset into the byte buffer to write
+   * @param v int value to write
+   */
+  public static void writeInt(ByteBuffer buf, int pos, int v) {
+    buf.put(pos ++, (byte) (0xff & (v >> 24)));
+    buf.put(pos ++, (byte) (0xff & (v >> 16)));
+    buf.put(pos ++, (byte) (0xff & (v >> 8)));
+    buf.put(pos, (byte) (0xff & v));
+  }
+
+  /**
+   * Writes a specific integer value (4 bytes) to the output stream.
+   *
+   * @param out output stream
+   * @param v integer value to write
+   * @throws IOException if an I/O error occurs. In particular, an <code>IOException</code> may be
+   *         thrown if the output stream has been closed.
+   */
+  public static void writeInt(OutputStream out, int v) throws IOException {
+    out.write((byte) (0xff & (v >> 24)));
+    out.write((byte) (0xff & (v >> 16)));
+    out.write((byte) (0xff & (v >> 8)));
+    out.write((byte) (0xff & v));
+  }
+
+  /**
+   * Writes a specific long value (8 bytes) to the output byte array at given offset.
+   *
+   * @param buf output byte array
+   * @param pos offset into the byte buffer to write
+   * @param v long value to write
+   */
+  public static void writeLong(byte[] buf, int pos, long v) {
+    checkBoundary(buf, pos, 8);
+    buf[pos ++] = (byte) (0xff & (v >> 56));
+    buf[pos ++] = (byte) (0xff & (v >> 48));
+    buf[pos ++] = (byte) (0xff & (v >> 40));
+    buf[pos ++] = (byte) (0xff & (v >> 32));
+    buf[pos ++] = (byte) (0xff & (v >> 24));
+    buf[pos ++] = (byte) (0xff & (v >> 16));
+    buf[pos ++] = (byte) (0xff & (v >> 8));
+    buf[pos] = (byte) (0xff & v);
+  }
+
+  /**
+   * Writes a specific long value (8 bytes) to the output stream.
+   *
+   * @param out output stream
+   * @param v long value to write
+   * @throws IOException if an I/O error occurs. In particular, an <code>IOException</code> may be
+   *         thrown if the output stream has been closed.
+   */
+  public static void writeLong(OutputStream out, long v) throws IOException {
+    out.write((byte) (0xff & (v >> 56)));
+    out.write((byte) (0xff & (v >> 48)));
+    out.write((byte) (0xff & (v >> 40)));
+    out.write((byte) (0xff & (v >> 32)));
+    out.write((byte) (0xff & (v >> 24)));
+    out.write((byte) (0xff & (v >> 16)));
+    out.write((byte) (0xff & (v >> 8)));
+    out.write((byte) (0xff & v));
+  }
+
+  /**
+   * Ensures there are len bytes from pos in buf.
```

Comment:
Ensures that the given buffer contains at least the given number of bytes after the given offset.